src/common/internet.rs

```diff
@@ -195,23 +204,35 @@ impl From<OnionAddressV3> for InetAddr {
     }
 }
 
+#[cfg(feature = "tor")]
+impl From<OnionAddressV2> for InetAddr {
+    #[inline]
+    fn from(addr: OnionAddressV2) -> Self {
+        InetAddr::TorV2(addr)
+    }
+}
+
 impl_try_from_stringly_standard!(InetAddr);
 impl_into_stringly_standard!(InetAddr);
 
 impl FromStr for InetAddr {
     type Err = String;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match IpAddr::from_str(s) {
-            Ok(ip_addr) => Ok(Self::from(ip_addr)),
+        match (
+            IpAddr::from_str(s),
+            OnionAddressV2::from_str(s),
+            OnionAddressV3::from_str(s),
```

Comment:
Can you pls change the order here: they are processed + optimized in that order by the compiler, and since Onion V3 addresses are much more abundant this may have a performance issue when processing large volumes of data (which may be a case for some apps like Bifrost or other types of nodes)

Reply:
Done.